avm/res/sql/server/database/main.bicep

```
@@ -2,146 +2,148 @@ metadata name = 'SQL Server Database'
 metadata description = 'This module deploys an Azure SQL Server Database.'
 metadata owner = 'Azure/module-maintainers'
 
+// START OF DATABASE PROPERTIES
+
 @description('Required. The name of the database.')
 param name string
 
 @description('Conditional. The name of the parent SQL Server. Required if the template is used in a standalone deployment.')
 param serverName string
 
+@description('Optional. The database SKU.')
+param sku databaseSkuType = {
+  name: 'GP_Gen5_2'
+  tier: 'GeneralPurpose'
+}
+
+@description('Optional. Time in minutes after which database is automatically paused. A value of -1 means that automatic pause is disabled.')
+param autoPauseDelay int = 0
+
+@description('Optional. Specifies the availability zone the database is pinned to.')
+param availabilityZone '1' | '2' | '3' | 'NoPreference' = 'NoPreference'
+
+@description('Optional. Collation of the metadata catalog.')
+param catalogCollation string = 'DATABASE_DEFAULT'
+
 @description('Optional. The collation of the database.')
 param collation string = 'SQL_Latin1_General_CP1_CI_AS'
 
-@description('Optional. The skuTier or edition of the particular SKU.')
-param skuTier string = 'GeneralPurpose'
+@description('Optional. Specifies the mode of database creation.')
+param createMode
+  | 'Default'
+  | 'Copy'
+  | 'OnlineSecondary'
+  | 'PointInTimeRestore'
+  | 'Recovery'
+  | 'Restore'
+  | 'RestoreExternalBackup'
+  | 'RestoreExternalBackupSecondary'
+  | 'RestoreLongTermRetentionBackup'
+  | 'Secondary' = 'Default'
+
+@description('Optional. The resource ID of the elastic pool containing this database.')
+param elasticPoolId string?
```

```suggestion
param elasticPoolResourceId string?
```

as per [spec](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr22---category-inputs---parametersvariables-for-resource-ids).